lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -343,6 +367,26 @@ class DefaultTextEditingStrategy implements TextEditingStrategy {
     _subscriptions.add(domElement.onKeyDown.listen(_maybeSendAction));
 
     _subscriptions.add(html.document.onSelectionChange.listen(_handleChange));
+
+    _subscriptions.add(domElement.onFocus.listen((_) {
+      bodyHasFocus = true;
+    }));
+
+    /// On default, do not blur the DOM element if the user goes to another
+    /// tab or minimizes the browser.
+    _subscriptions.add(domElement.onBlur.listen((_) {
+      // Cancel previous timer if exists.
+      pageVisibilityTimer?.cancel();
+      pageVisibilityTimer = Timer(_delayBeforePageVisibilityTimer, () {
+        if (bodyHasFocus) {
+          /// Focus is still on the body. Continue with blur.
+          owner.sendTextConnectionClosedToFlutterIfAny();
+        } else {
+          // Refocus.
+          domElement.focus();
+        }
+      });
```

Comment:
Thanks for the suggestion! After the suggestion I tried to listen blur on window with capture phase enabled, but it didn't work.
Here is the problem:
"If an input element is in focus and the window is blurred (let's say by clicking on another tab)" the event getting triggered has still have target==input and currentTarget==input. This behavior is identical to the behavior "If an input element is in focus and a random element on the page is triggered".  Both events will still be captured on the window and both will have the same targets.

Good news is since our goal is to remove the timer and reduce the complexity, I came up with other solutions*:
1) html.document.hasFocus returns if the body has focus on Safari and Chrome (both mobile and desktop) in these browsers we can use this values instead of checking the timer.
2) (1) does not work in firefox. However document.activeElement always equals to domElement if the window loses focus on Firefox. So in the FirefoxTextEditingStrategy I'll be using this approach.

(*) tested on: Chrome on Linux, Firefox on Linux, Chrome on Android, Safari on IOS, Safari on MacOS

PHAL once I push the new version of the PR. I am fixing the comments now :)